pkg/state/appender.go

```diff
@@ -794,10 +786,14 @@ func (a *txAppender) appendBlock(params *appendBlockParams) error {
 	if ssErr := a.stor.snapshots.saveSnapshots(params.block.BlockID(), currentBlockHeight, bs); ssErr != nil {
 		return ssErr
 	}
+
+	// clean up legacy state hash records with zero diffs
+	a.stor.balances.filterZeroDiffsSHOut(blockID)
```

Comment:
It's important to do these actions in `snapshotApplier` because only this entity (mostly) can change the state in the light node mode.